### ISSUE TITLE
Konsole is slow to render large screens. Speed, Speed, Speed.

### ISSUE THREAD

**Issue by goblinfactory** [goblinfactory/konsole#32]
When rendering large screens with lots of layout Konsole can take up to 1 or 2 seconds to render a whole layout. While ok for the original use case of being able to help visualise output from scripts with parts running in parallel, its not OK to use to create a real application or "for production" utility.

**Comment by goblinfactory** [comment]
busy with this now, so far tests are looking good.

**Comment by goblinfactory** [comment]
pushed draft "partial implementation to branch" https://github.com/goblinfactory/konsole/tree/%2332-speed

**Comment by goblinfactory** [comment]
the following code
```csharp

         using (var writer = new HighSpeedWriter())
            {
                var window = new Window(writer);
                //var window = new Window();

                var consoles = window.SplitRows(
                        new Split(4, "heading", LineThickNess.Single),
                        new Split(10),
                        new Split(0),
                        new Split(4, "status", LineThickNess.Single)
                ); ; ;

                var headline = consoles[0];
                var contentTop = consoles[1];
                var contentBottom = consoles[2];
                var status = consoles[3];

                var longText = "Let's see if these do? heres more text to see if this eventually wraps? text to see if this eventually wraps?text to see if this eventually wraps?";

                var menu = contentTop.SplitLeft("menu");
                var content = contentTop.SplitRight("content");

                var splits = contentBottom.SplitColumns(
                        new Split(20, "menu2"),
                        new Split(0, "content2"),
                        new Split(20, "content3")
                    );
                var menu2 = splits[0];
                menu2.WriteLine(longText);
                var content2 = splits[1];
                var content3 = splits[2];

                content2.WriteLine(longText);
                content3.WriteLine(longText);

                headline.Write("my headline");
                content.WriteLine("content goes here");

                // ascii characters dont print when printed as unicode....mmm, so close? 

                content.WriteLine("these lines ╢╖╣║╗╟  don't print!");
                
                // content doesnt wrap??
                content.WriteLine("Let's see if these do? heres more text to see if this eventually wraps? text to see if this eventually wraps?text to see if this eventually wraps?");

                menu.WriteLine("Options A");
                menu.WriteLine("Options B");
                menu.WriteLine("Vvery very long option C that should wrap quite a bit! Vvery very long option C that should wrap quite a bit! Vvery very long option C that should wrap quite a bit!");
                menu.WriteLine("Options D");

               // sidebar.WriteLine("20% off all items between 11am and midnight tomorrow!");

                status.Write("System offline!");
                writer.Flush();
                menu2.WriteLine(ConsoleColor.Red, "press any key to print a bunch of text into content2, press 'q' to quit");

                char key = 'x';
                int color = 0;
                while(key != 'q')
                {
                    content2.WriteLine((ConsoleColor)(color++ % 15), longText);
                    writer.Flush();
                    key = Console.ReadKey().KeyChar;

                }
                
            }
```
creates this ... (and does it it really really quickly ) will publish speed increases shortly.

![HighSpeedWriter](https://user-images.githubusercontent.com/1073487/70377588-17ef0b80-190e-11ea-9050-f24a7d6626cd.PNG)

**Comment by goblinfactory** [comment]
closing this issue. Speed update is available in version 5.0 >
